app/controllers/smart_proxies_controller.rb

```diff
@@ -38,11 +38,11 @@ def refresh
   end
 
   def ping
-    requested_data(:version)
+    requested_data(ProxyStatus::Base.new(@smart_proxy), :version)
```

Comment:
i think this should be converted to a before filter? then you would initialize it only once (in case you needed it another place later on)